include/ChocAn/core/utils/account_builder.hpp

```diff
@@ -0,0 +1,108 @@
+/* 
+File: account_builder.hpp
+
+Brief:  Account builder performs exception handling and guides the user
+        through the process of creating an account.
+
+Authors: Daniel Mendez 
+         Alex Salazar
+         Arman Alauizadeh 
+         Alexander DuPree
+         Kyle Zalewski
+         Dominique Moore
+
+https://github.com/AlexanderJDupree/ChocAn
+ 
+*/
+
+#ifndef CHOCAN_ACCOUNT_BUILDER_HPP
+#define CHOCAN_ACCOUNT_BUILDER_HPP
+
+#include <ChocAn/core/data_gateway.hpp>
+#include <ChocAn/core/entities/account.hpp>
+#include <ChocAn/view/terminal_input_controller.hpp>
+#include <stack>
+#include <optional>
+
+enum field_types{current_type,current_first,current_last,current_street,current_city,current_state,current_zip};
+
+struct Build_Instructions{
+
+    Build_Instructions(){};
+    
+    const std::string account_type   
```

Comment:
The explicit formatting of these strings break design principles. Namely, the domain layer should zero knowledge that it is being used in a console application. This explicit formatting should be handled in the presentation layer with the resource loader. 

So maybe it would be better to convert these to an enumerated type or a variant since you seem to be using these strings as a basis of comparison in your `set_current_field` function. And then in the resource loader, depending on the type of the build instruction we generate the correct string.

Reply:
ill be using a variant for tracking what build phase we are on during the building process